tree/tree/src/TChain.cxx

```diff
@@ -2303,6 +2304,47 @@ void TChain::ResetAfterMerge(TFileMergeInfo *info)
    TTree::ResetAfterMerge(info);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Save TChain as a C++ statement(s) on output stream out
+
+void TChain::SavePrimitive(std::ostream &out, Option_t *option)
+{
+   TList chains;
+   chains.Add(this);
+
+   out << "   TChain *" << fName.Data() << " = new TChain(\"" << fName.Data() << "\");" << std::endl;
+
+   for (TObject *el : *fFriends) {
+      TChain *ch = (TChain *)((TFriendElement *)el)->GetTree();
+      chains.Add(ch);
+      out << "   TChain *" << ch->GetName() << " = new TChain(\"" << ch->GetName() << "\");" << std::endl;
```

Comment:
Unfortunately the name of TChain is not necessarily a valid C++ symbol (see other SavePrimitive implementations).

Reply:
Will do it as in TH1::SavePrimitive(). By the way TF1::SavePrimitive() which I based on has this bug.

Reply:
Hi,
Please, have a look at the new commits. I have changed code as you requested and also implemented a test.